internal/paascharm/gogen.go

```diff
@@ -0,0 +1,81 @@
+package paascharm
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path/filepath"
+	"text/template"
+)
+
+// CreateGoStructs reads the charmcraft.yaml file from the charmcraftFile location
+// and outputs the Go file with the proper structs in the outputFile location,
+// using as package name packageName. It will override the outputFile file if
+// it exists and create all the parent directories if missing.
+func CreateGoStructs(charmcraftFile string, packageName string, outputFile string) (err error) {
+	yamlFile, err := os.ReadFile(charmcraftFile)
+	if err != nil {
+		return fmt.Errorf("cannot read charmcraft.yaml file: %v", err)
+	}
+
+	charmcraftConfig, err := ParseCharmcraftYaml(yamlFile)
+	if err != nil {
+		return fmt.Errorf("error parsing charmcraft.yaml file: %v", err)
+	}
+
+	goStructInfo, err := NewGoStructsData(packageName, charmcraftConfig)
+	if err != nil {
+		return fmt.Errorf("cannot create go structs data: %v", err)
+	}
+
+	goStructs, err := GenerateGoStructs(goStructInfo)
+	if err != nil {
+		return fmt.Errorf("cannot generate go structs info: %v", err)
+	}
+
+	err = os.MkdirAll(filepath.Dir(outputFile), 0755)
+	if err != nil {
+		return fmt.Errorf("cannot create parent directories for output file %s: %v", outputFile, err)
+	}
+
+	if _, err := os.Stat(outputFile); err == nil {
+		log.Printf("file %s already exists and will be overridden\n", outputFile)
+	}
+
+	err = os.WriteFile(outputFile, goStructs, 0644)
+	if err != nil {
+		return fmt.Errorf("cannot write output file %s: %v", outputFile, err)
+	}
+	log.Printf("configuration written to file: %s\n", outputFile)
+
+	return
+}
+
+//go:embed go.tmpl
+var GoTemplate string
```

Comment:
Nice use of `go:embed`!

However, `GoTemplate` doesn't need to be exported (capitalised) -- it's an internal implementation detail of this package. Could rename to `goTemplate` or similar.